plugins/scaffolder/src/api.ts

```diff
@@ -124,26 +70,33 @@ export class ScaffolderClient implements ScaffolderApi {
     this.useLongPollingLogs = options.useLongPollingLogs ?? false;
   }
 
-  async getIntegrationsList(options: { allowedHosts: string[] }) {
-    return [
+  async getIntegrationsList(
+    options: ScaffolderGetIntegrationsListOptions,
+  ): Promise<ScaffolderGetIntegrationsListResponse> {
+    const integrations = [
       ...this.scmIntegrationsApi.azure.list(),
       ...this.scmIntegrationsApi.bitbucket.list(),
       ...this.scmIntegrationsApi.github.list(),
       ...this.scmIntegrationsApi.gitlab.list(),
     ]
       .map(c => ({ type: c.type, title: c.title, host: c.config.host }))
       .filter(c => options.allowedHosts.includes(c.host));
+
+    return {
+      integrations,
+    };
   }
 
   async getTemplateParameterSchema(
-    templateName: EntityName,
+    templateRef: string,
   ): Promise<TemplateParameterSchema> {
-    const { namespace, kind, name } = templateName;
+    const { namespace, kind, name } = parseEntityRef(templateRef);
```

Comment:
do note that this now requires that this is a FULL ref - if you want support for passing in just the name etc, you need to give this a defaultKind and defaultNamespace in the options object